apps/agent-service/src/repositories/agent-service.repository.ts

```diff
@@ -418,4 +435,33 @@ export class AgentServiceRepository {
       throw error;
     }
   }
+
+  async getLedgerByNameSpace(indyNamespace: string): Promise<{
+    id: string;
+    createDateTime: Date;
+    lastChangedDateTime: Date;
+    name: string;
+    networkType: string;
+    poolConfig: string;
+    isActive: boolean;
+    networkString: string;
+    nymTxnEndpoint: string;
+    indyNamespace: string;
+    networkUrl: string;
+}> {
+    try {
+      if (indyNamespace) {
+        const ledgerDetails = await this.prisma.ledgers.findFirst({
```

Comment:
Please avoid using `findFirst`. I will return the 1st value if the namespace does not match